### PR TITLE
Fix bugs spotted by QA

### PR DESCRIPTION
**Pull request by polishq**
**What problem is this solving?**

Fixes many bugs spotted by QA:

- https://vtex-dev.atlassian.net/browse/WORDPRESS-61 (Pagination wasn't working in some components)
- https://vtex-dev.atlassian.net/browse/WORDPRESS-62 (Search terms were not decoded in the breadcrumb component)
- https://vtex-dev.atlassian.net/browse/WORDPRESS-63 (Changing "posts per page" wasn't working)
- https://vtex-dev.atlassian.net/browse/WORDPRESS-64 (User feedback was unclear when filtering by a tag with no posts)
- https://vtex-dev.atlassian.net/browse/WORDPRESS-65 (Filter by date range was missing some results)

along with a few other issues:

- adds try/catch to all GraphQL resolvers, so that errors are caught and logged instead of being sent to the frontend
- fixes broken category links (`page` param was missing)
- replaces all hardcoded english strings with intl messages
- fixes various SonarCloud issues
- improves typings
- removes unnecessary props

UPDATE: Also updated many resolvers because we were sending arrays to the Wordpress API (i.e. `tags=[6, 8]`) when they should be comma-separated lists (i.e. `tags=6,8`) and fixed an issue where `customDomain` wasn't being passed to the tagResolver correctly.

**How should this be manually tested?**

Linked here: https://arthur--worldwidegolf.myvtex.com/insider
and here: https://wp324--productusqa.myvtex.com/blog
and here (for the `customDomain` problem): https://arthur--prodequa.myvtex.com/producto-de-prueba-actualizado/p

### CODE REVIEW

**Review by Arturo777 — APPROVED** [review]
LGTM

File: react/components/WordpressPost.tsx [hunk 1]
```diff
@@ -215,7 +116,7 @@ const WordpressPostInner: FunctionComponent<WordpressPostInnerProps> = (
   const formattedDate = dateObj.toLocaleDateString(locale, dateOptions)
 
   const productIds = tags
-    .filter((tag: WPTag) => tag.name && tag.name.includes('prod-'))
+    .filter((tag: WPTag) => tag?.name?.includes('prod-'))
```
Comment (whc105) [inline react/components/WordpressPost.tsx:119]: Maybe we can do a .reduce here to combine the .filter and .map so we only need to iterate once

Comment (polishq) [reply react/components/WordpressPost.tsx]: Good call, done 👍

**Review by whc105 — APPROVED** [review]
lgtm

**Review by whc105 — APPROVED** [review]

**Review by Arturo777 — DISMISSED** [review]
LGTM

**Review by Arturo777 — APPROVED** [review]
LGTM

**Pull request merged**